### ISSUE TITLE
Support for Technicolor TG799vac VANT-R

### ISSUE THREAD

**Issue by bufce2010** [kevdagoat/hack-technicolor#42]
**Hello Everyone,**

I'm new here, and I hope I could find help.

**.Regarding this router hacking tutorial:**
https://hack-technicolor.readthedocs.io/en/latest/Hack%20Type%201&2/

**. I'm trying to flash "Telia TechnoColor" modem, I copied those information before starting the hacking tutorial:-**
Product Vendor: Technicolor
Product Name: MediaAccess TG799vac
Software Version: 15.3
Firmware Version: 15.51.6436-1361003-20160202112931
Hardware Version: VANT-R
NTP servers: ntp1.rgw.telia.se

**. I walked through the process to the "Bank Planning":- (I'm not an advanced user)**

**The Tutorial says that The optimal bank plan looks like this:**
/proc/banktable/active
bank_1
/proc/banktable/booted
bank_2
**. I didn't get the same and I don't know what I did wrong? and how can I fix it? and what to do next?.**

/proc/banktable/passiveversion
Unknown
/proc/banktable/activeversion
15.51.6436-1361003-20160202112931-ebb9150c3d7ffdd1bb106bc3629d993f5fe2443d
/proc/banktable/inactive
bank_2
/proc/banktable/active
bank_1
/proc/banktable/notbooted
bank_2
/proc/banktable/booted
bank_1

**Comment by LuKePicci** [comment]
Thank you, I just found an error in the guide. The underlined part should have been `bank_2`, however the actual command was already correct
![image](https://user-images.githubusercontent.com/8722358/64346909-4994c580-cff3-11e9-8472-8c1de37513d5.png)

 Which bank planning section were you following? With or w/out fw upgrade?

**Comment by bufce2010** [comment]
> Thank you, I just found an error in the guide. The underlined part should have been `bank_2`, however the actual command was already correct
> ![image](https://user-images.githubusercontent.com/8722358/64346909-4994c580-cff3-11e9-8472-8c1de37513d5.png)

I've just copied and pasted the commands.
I understand few of them but most others goes over my head.
 
> Which bank planning section were you following? With or w/out fw upgrade?

I followed the "without firmware upgrade".

**Comment by LuKePicci** [comment]
You wrote:
> The Tutorial says that The optimal bank plan looks like this:

however that part was only int the "with firmware upgrade" section. Are you sure you didn't switched by mistake from a path to another?

Anyway, I don't know your original state, but from your current situation you can just continue reading from here https://hack-technicolor.readthedocs.io/en/latest/Hack%20Type%201&2/#bank-planning-without-firmware-upgrade and go on as usual. As said in the guide, you will reach the optimal bank plan at the end.

**Comment by bufce2010** [comment]
> You wrote:
> 
> > The Tutorial says that The optimal bank plan looks like this:
> 
> however that part was only in the "with firmware upgrade" section. Are you sure you didn't switch by mistake from a path to another?


Oh No!
I must have mixed between them by mistake


> Anyway, I don't know your original state, but from your current situation you can just continue reading from here https://hack-technicolor.readthedocs.io/en/latest/Hack%20Type%201&2/#bank-planning-without-firmware-upgrade and go on as usual. As said in the guide, you will reach the optimal bank plan at the end.

don't I need to modify a specific commands since bank_2 and bank_1 are switched?

**Comment by LuKePicci** [comment]
Nope, every bank planning section is written to handle every possible starting situation, so even your current one should work just fine

**Comment by bufce2010** [comment]
> Nope, every bank planning section is written to handle every possible starting situation, so even your current one should work just fine

sorry but the router is still locked with telia firmware.
would I get to unlock it at the end of the tutorial?

**Comment by LuKePicci** [comment]
I'm not sure if I understand. We say the router is "rooted" when you have root access to the shell. If you have access to the root shell but you didn't complete the bank planning stage you can restart reading from that stage and continue to the end, otherwise if you lost root access by executing the wrong block of commands you need to restart the guide from the beginning.
I assumed you still have root access since you were able to print out current banks state.

Any kind of "unlocking" or "modding" will be done as extra steps after you complete the rooting tutorial.

**Comment by bufce2010** [comment]
I appreciate your help.
I see now that I have root access and I'm supposed to be able to flash an unlocked fw,
but the router doesn't recognise the usb flash disk as I try to push the fw using the following command: 

```
cp vant-f_CRF683-17.2.188-820-RA.rbi /tmp
```
it says
cp: can't stat 'vant-f_CRF683-17.2.188-820-RA.rbi': No such file or directory

**Comment by LuKePicci** [comment]
Ok, take a moment to understand your situation. There no "unlocked firmware". We use to root and therefore unlock things in existing firmwares. You have a VANT-R, so you could only flash firmwares for VANT-R, not VANT-F. Every firmware for VANT-R can be rooted (and therefore "unlocked"), as long as there exists another firmware for VANT-R (like your current firmware) which is rootable by itself.

The cp command give you error because when you log into SSH you are not inside the usb flashdisk already. Don't care about this for now.

Avoid following the "with upgrade" path for now, we need something more to let you install the update. Complete the "bank planning without upgrade" stage, continue reading, stop as soon as you are told to start "unlocking functionalities". Then read the following part of this message.

I would like to ask you something about your model such that we could add it to the wiki as well:
- Are you from Telia Sweden or Finland?
- How did you get root access? AutoflashGUI?
- Which is the commercial name your ISP gives to this device? Maybe something like "Best UltraGigaModem Ever" or anything like that, or just TG799vac? Even if it's not in English it's fine
- I have newer firmwared for your VANT-R, I will link them to the wiki if you give me enough info to decently support your device
- I don't have a copy of your current firmware. I would like to get a dump of your current firmware (may be useful! newer ones may require this to be unlocked). Please follow the guide [here](https://hack-technicolor.readthedocs.io/en/latest/Resources/#backuprestore-bit-for-bit-dumps) to get a dump of every partition. Maybe I would also be able to also get the RBI for it but I need someone on the telia network who is capable of accessing telia file server from mgmt vlan. It's way easier if you download it from the router itself as it is already connected to Telia. Let's skip this for now, we need to consolidate the support for your model first. 
- We still don't know the OSCK for your VANT-R. OSCK is needed to allow you and everybody decrypting .rbi firmwares for VANT-R. I already asked you to make dumps of every partition, so you should have got a dump for eripv2 partition as well.  Read [here](https://github.com/pedro-n-rocha/secr) about how to get it.

Take your time.

**Comment by bufce2010** [comment]
>Are you from Telia Sweden or Finland?

**I'm in Egypt right now, and I wanted to connect to an ISP called "WE" previously called "TE-DATA". 
the modem is firmware locked to Telia Sweden**

>How did you get root access?

**I used AutoflashGUI**


>Which is the commercial name your ISP gives to this device? Maybe something like "Best UltraGigaModem Ever" or anything like that, or just TG799vac? Even if it's not in English it's fine
>I have newer firmwared for your VANT-R, I will link them to the wiki if you give me enough info to...... 
We still don't know the OSCK for your.... 


**I forgot the router connected to my pc at the office. I'll try to get on TeamViewer via my mobile phone, otherwise I'll be have to wait till Saturday**

**Comment by LuKePicci** [comment]
> I'm in Egypt right now, and I wanted to connect to an ISP called "WE" previously called "TE-DATA".
> the modem is firmware locked to Telia Sweden
That's fine
> I used AutoflashGUI
great
> forgot the router connected to my pc at the office. I'll try to get on TeamViewer via my mobile phone, otherwise I'll be have to wait till Saturday
No problem 👍

**Comment by bufce2010** [comment]
**Sorry I took that long to reply as I had to extend my vacation.**

> Which is the commercial name your ISP gives to this device? Maybe something like "Best UltraGigaModem Ever" or anything like that, or just TG799vac? Even if it's not in English it's fine

**I Hope those lines provide the answer:-
Product Vendor: Technicolor
_Product Name: MediaAccess TG799vac_
Software Version: 15.3
Firmware Version: 15.51.6436-1361003-20160202112931
Hardware Version: VANT-R
Serial Number: "let me know if you need it"
MAC Address: "let me know if you need it"
NTP servers: ntp1.rgw.telia.se**

> I have newer firmwares for your VANT-R, I will link them to the wiki if you give me enough info to properly support your device

**That would be great, thanks.**
**happy to help, I have teamviewer installed if there is something I can't do right**

> I don't have a copy of your current firmware. I would like to get a dump of your current firmware (may be useful! newer ones may require this to be unlocked). Please follow the guide here to get a dump of every partition. Upload all of them in a zip into my private MEGA folder here. Maybe I would also be able to also get the RBI for it but I need someone on the telia network who is capable of accessing telia file server from mgmt vlan. It's way easier if you download it from the router itself as it is already connected to Telia. Let's skip this for now, we need to consolidate the support for your model first.

**OK I ran the first command and here is the result**

- **root@OpenWrt:~# md5sum /dev/mtd***

62d17d4a3ccad3cac749675f2ef08dd1  /dev/mtd0
eec571166b8f46ea118e867a459cc3af  /dev/mtd0ro
4258506f9de2fab14292883fd095ce7e  /dev/mtd1
4258506f9de2fab14292883fd095ce7e  /dev/mtd1ro
262617ae73d0f4abdd7f9678cc9de515  /dev/mtd2
262617ae73d0f4abdd7f9678cc9de515  /dev/mtd2ro
d4f2a72192dced2054d486873a405243  /dev/mtd3
d4f2a72192dced2054d486873a405243  /dev/mtd3ro
fe34ce8347e7489744fb154f0b1d7558  /dev/mtd4
fe34ce8347e7489744fb154f0b1d7558  /dev/mtd4ro
c8ee79e3c7f792177aa3d0bdcbd5ce7a  /dev/mtd5
c8ee79e3c7f792177aa3d0bdcbd5ce7a  /dev/mtd5ro
367e7c2b842d90c7d41f2d82b87c7ae1  /dev/mtd6
367e7c2b842d90c7d41f2d82b87c7ae1  /dev/mtd6ro
44d5b591392bc0d16d74b1b076f17962  /dev/mtd7
44d5b591392bc0d16d74b1b076f17962  /dev/mtd7ro
la627cb93d7a1c68bb746b86fe6a598e5  /dev/mtdblock0
4258506f9de2fab14292883fd095ce7e  /dev/mtdblock1
4fb5215fcb0180f374b1a8d3a9a50c8a  /dev/mtdblock2
d4f2a72192dced2054d486873a405243  /dev/mtdblock3
fe34ce8347e7489744fb154f0b1d7558  /dev/mtdblock4
c8ee79e3c7f792177aa3d0bdcbd5ce7a  /dev/mtdblock5
367e7c2b842d90c7d41f2d82b87c7ae1  /dev/mtdblock6
d41d8cd98f00b204e9800998ecf8427e  /dev/mtdblock7

- **then listed connected usb drive**
**root@OpenWrt:~# ls -la /mnt/usb**

drwxr-xr-x    3 root     root             0 Jan  3 18:50 .
drwxr-xr-x    1 root     root             0 Jan  1 03:09 ..
drwxrwxrwx    6 root     root          8192 Jan  1 01:00 Kingston_DT101G2_1_3814
-  **But I can't figure the correct syntax of the command:**
`root@OpenWrt:~# dd if=/dev/mtd<X> of=/tmp/mtd<X>.dump`
**With what should I replace < X >?**
-  **mv /tmp/mtdX.dump /mnt/usb/< usb-path >/**
also what will be the < usb-path > in this case

>We still don't know the OSCK for your VANT-R. OSCK is needed to allow you and everybody decrypting .rbi firmwares for VANT-R. I already asked you to make dumps of every partition, so you should have got a dump for eripv2 partition as well. Read here about how to get it. If you don't know how to run python scripts just drop your dmesg output inside the same MEGA folder of mine.

**and I have no idea**

**Comment by LuKePicci** [comment]
As you see, usb path is Kingston_DT101G2_1_3814 so: `/mnt/usb/Kingston_DT101G2_1_3814/`

**Comment by bufce2010** [comment]
**something went wrong**
![image](https://user-images.githubusercontent.com/50169186/64488648-9d4d1c00-d24a-11e9-821b-93a3f44511e5.png)

**Comment by LuKePicci** [comment]
Oh yeah, nothing to worry about, just use `dd if=/dev/mtd<X> of=/mnt/usb/Kingston_DT101G2_1_3814/mtd<X>.dump` directly, there is not enough space in ram (/tmp) to store the big mtd0 dump.

**Comment by bufce2010** [comment]
> Oh yeah, nothing to worry about, just use
> `dd if=/dev/mtd<X> of=/mnt/usb/Kingston_DT101G2_1_3814/mtd<X>.dump`
> directly, there is not enough space in ram (/tmp) to store the big mtd0 dump.
> 
> Please note, it will take some minutes to dump the full flash

**Done uploading them**

**Comment by bufce2010** [comment]
```
root@OpenWrt:~# ls -la /mnt/usb/Kingston_DT101G2_1_3814/*.ko
-rwxrwxrwx    1 root     root         10638 Aug 19  2019 /mnt/usb/Kingston_DT101G2_1_3814/lime.arm                                                            .3.4.ko
-rwxrwxrwx    1 root     root          4531 Aug 19  2019 /mnt/usb/Kingston_DT101G2_1_3814/r2secr.a                                                            rm.3.4.ko
-rwxrwxrwx    1 root     root          4150 Aug 19  2019 /mnt/usb/Kingston_DT101G2_1_3814/r2secr.a                                                            rm.4.1.ko
-rwxrwxrwx    1 root     root          3985 Aug 19  2019 /mnt/usb/Kingston_DT101G2_1_3814/r2secr.m                                                            ips.2.6.ko
-rwxrwxrwx    1 root     root          4266 Aug 19  2019 /mnt/usb/Kingston_DT101G2_1_3814/r2secr.m                                                            ips.3.4.ko
-rwxrwxrwx    1 root     root         37183 Aug 19  2019 /mnt/usb/Kingston_DT101G2_1_3814/ripdrv.a                                                            rm.3.4.ko

root@OpenWrt:/mnt/usb# cd /mnt/usb/Kingston_DT101G2_1_3814/
root@OpenWrt:/mnt/usb/Kingston_DT101G2_1_3814# uname -a
Linux OpenWrt 3.4.11-rt19 #1 SMP PREEMPT Tue Dec 15 14:39:05 CET 2015 mips GNU/Linux

```

**Comment by LuKePicci** [comment]
Great, now use the last one: 
`insmod /mnt/usb/Kingston_DT101G2_1_3814/ripdrv.arm.3.4.ko`
then run `dmesg` immediately after. If you look at last few lines there will be your personal ECKey printed. Use that ECK plus your mtd5 (eripv2) dump you made with the eripv2.py script to get the VANT-R OSCK, or drop a txt file with your ECK in my MEGA and I will run the script for you. Note, the ECK is personal, the OSCK is equale for all VANT-R.

**Comment by wuseman** [comment]
I don't know if this will help you, but VBNT-r firmwares can be found [here](https://github.com/wuseman/TG799VAC-XTREME-17.2-MINT/issues/5#issuecomment-445488962) for your modem.

**Comment by LuKePicci** [comment]
@wuseman could you get the RBI for this old 15.x version @bufce2010 has on its device? It should be available at `http://131.116.22.230:80/15516436r1361003closed.rbi` from Telia mgmt private vlan network (vid=254, dhcp).

> I don't know if this will help you, but VANT-R firmwares can be found here for your modem.

Yeah, that's what I was thinking of, I also have another one older 17.2 in the middle. You also recall me to add the VBNT-H firmware I have to the repo page as well.

**Comment by wuseman** [comment]
> @wuseman could you get the RBI for this old 15.x version @bufce2010 has on its device? It should be available at `http://131.116.22.230:80/15516436r1361003closed.rbi` from Telia mgmt private vlan network (vid=254, dhcp).
> 
> > I don't know if this will help you, but VANT-R firmwares can be found here for your modem.
> 
> Yeah, that's what I was thinking of, I also have another one older 17.2 in the middle. You also made me think I still have to add the VBNT-H firmware I have to the repo page as well.

Unfortunately not, my firmware is locked 17.04... and I can't login anymore to get root acces after their last upgrade. Maybe the guy who find these can help if he reading this

How do figure out the filename? I have alot of logs from that IP and I event found the key for these firmwares.

Some examples:
set osgi.@bundle_url[0].url=http://131.116.22.230/repo/osgi5_17_2/TSBoot.jar
cwmpd.cwmpd_config.interface6='wan6'
cwmpd.cwmpd_config.ip_preference='v4_only'
cwmpd.cwmpd_config.periodicinform_time='0'
cwmpd.cwmpd_config.activenotification_throttle='0'
cwmpd.cwmpd_config.connectionrequest_throttle_time='0'
cwmpd.cwmpd_config.connectionrequest_throttle_number='1'
cwmpd.cwmpd_config.connectionrequest_path='8pVIaHLICt3cbRBD'
cwmpd.cwmpd_config.ssl_castore='/etc/ssl/certs/'
cwmpd.cwmpd_config.ssl_datecheck='1'
cwmpd.cwmpd_config.ssl_ciphersuite='ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:!DSS'
cwmpd.cwmpd_config.namespace_version='1-2'
cwmpd.cwmpd_config.backoff_minwait='5'
cwmpd.cwmpd_config.backoff_multiplier='2000'
cwmpd.cwmpd_config.datamodel='InternetGatewayDevice'
cwmpd.cwmpd_config.connectionrequest_url='http://10.144.236.176:51005/8pVIaHLICt3cbRBD'

If I understand this correct, http://10.144.236.176:51005/8pVIaHLICt3cbRBD would be http://10.144.236.176:51005/8pVIaHLICt3cbRBD.rbi?

Sorry if oft, mine is VBNT-w atm .

**Comment by LuKePicci** [comment]
Please open a new issue for "Telia firmware URL guessing", I will be happy to explain there what I usually do to find them.

**Comment by wuseman** [comment]
Sure, thanks

**Comment by LuKePicci** [comment]
Back on topic, may be you know if telia calls this old VANT-R something else then just "Technicolor TG799vac"? Did they use any supercool name on the box?

**Comment by bufce2010** [comment]
> Great, now use the last one:
> `insmod /mnt/usb/Kingston_DT101G2_1_3814/ripdrv.arm.3.4.ko`

root@OpenWrt:/mnt/usb/Kingston_DT101G2_1_3814#
`insmod /mnt/usb/Kingston_DT101G2_1_3814/ripdrv.arm.3.4.ko`
insmod: can't insert '/mnt/usb/Kingston_DT101G2_1_3814/ripdrv.arm.3.4.ko': invalid module format

> Back on topic, may be you know if telia calls this old VANT-R something else then just "Technicolor TG799vac"? Did they use any supercool name on the box?

It came without the box :( 
![image](https://user-images.githubusercontent.com/50169186/64510305-24999e80-d2e2-11e9-9dbc-bf1056487c5a.png)

**Comment by bufce2010** [comment]
> I don't know if this will help you, but VANT-R firmwares can be found [here](https://github.com/wuseman/TG799VAC-XTREME-17.2-MINT/issues/5#issuecomment-445488962) for your modem.

Thanks for sharing, I have them already from the same comment. I've done a lot of search before posting.

**Comment by bufce2010** [comment]
> @bufce2010 Awesome. The correct module was that one for mips, I forget to tell you.

**great, what is next?
I've installed Python3**
![image](https://user-images.githubusercontent.com/50169186/64515635-392f6400-d2ed-11e9-89e7-79b067f94451.png)

> Commercial name from Telia could be this one: "Telia Tradios router", here is its box

**yeah, that's also printed on the router like in the photo** 
![image](https://user-images.githubusercontent.com/50169186/64515882-b529ac00-d2ed-11e9-89af-22136acba75f.png)

> Now I ask you to wait a bit, I have to push some changes here and there to support this board. I will be back here once you are ready to go with firmware upgrade.

**ok great job.
I'll be here when you come back.**

**Comment by flywire** [comment]
> Commercial name ... "Telia Tradios router"

No vendor standard but "Telia - Tradios router" for consistency in the guide.

**Comment by bufce2010** [comment]
@LuKePicci 
**I've dropped the "mtd5.dd" file in your Mega Folder**

**Comment by LuKePicci** [comment]
Thanks, now working on it. The OSCK I extracted is working as expected. The ECK is unique for your piece of hardware but it is not a problem to share unless you also give the mtd5 to anyone. Just in case, I also edited your previous comment to hide the ECK too.

**Comment by bufce2010** [comment]
> Thanks, now working on it. The OSCK I extracted is working as expected. The ECK is unique for your piece of hardware but it is not a problem to share unless you also give the mtd5 to anyone. Just in case, I also edited your previous comment to hide the ECK too.

Awesome, You are on of your kind :) (I Hope that's appropriate)

**Comment by LuKePicci** [comment]
This is now completely supported from https://github.com/kevdagoat/hack-technicolor/commit/31978c315527382e8d7ecfb2d9948cce922514f7

@bufce2010 I still need to prepare a draft for the "Safe Firmware Upgrade" page you need to perform the upgrade. The current wiki only supports on-the-fly upgrades during initial rooting tutorial, so I ask you to wait a bit further for the firmware update.

Meanwhile, I noticed you managed to move to the optimal bank plan (well done). Of course your current firmware is not supported by https://github.com/Ansuel/tch-nginx-gui, maybe @ansuel could add support for newer versions but you will need to do a little bit of advanced testing and debugging with support from him. I can't give you any further help from myself if you would like to go for the custom GUI. Having root access you are now able to configure this device for your isp by editing configuration files inside the /etc/config/ folder. You can browse all files from WinSCP (use SSH user/password to login).

I can give you some hints about this if you tell me which kind of settings your "WE" isp requires. In general, you will need to disable wansensing inside /etc/config/wansensing, configure your pppoe or dhcp internet connection (with credentials, if any) inside /etc/config/network and your voip inside /etc/config/mmpbx* (with some adjustments inside /etc/config/mwan and /etc/config/dhcp). You will need to change as much options as much your ISP settings differs from Telia ones.

Let me close this issue as rooting support for VANT-R is now complete. Please, open a new issue for "Help configuring VANT-R for WE" if needed, sharing every useful knowledge you already have about your ISP settings.

Thank you for your help.

**Comment by bufce2010** [comment]
> This is now completely supported from [31978c3](https://github.com/kevdagoat/hack-technicolor/commit/31978c315527382e8d7ecfb2d9948cce922514f7)
> 
> @bufce2010 I still need to prepare a draft for the "Safe Firmware Upgrade" page you need to perform the upgrade. The current wiki only supports on-the-fly upgrades during initial rooting tutorial, so I ask you to wait a bit further for the firmware update.
> 
> Meanwhile, I noticed you managed to move to the optimal bank plan (well done). Of course your current firmware is not supported by https://github.com/Ansuel/tch-nginx-gui, maybe @Ansuel could add support for newer versions but you will need to do a little bit of advanced testing and debugging with support from him. I can't give you any further help from myself if you would like to go for the custom GUI. Having root access you are now able to configure this device for your isp by editing configuration files inside the /etc/config/ folder. You can browse all files from WinSCP (use SSH user/password to login).
> 
> I can give you some hints about this if you tell me which kind of settings your "WE" isp requires. In general, you will need to disable wansensing inside /etc/config/wansensing, configure your pppoe or dhcp internet connection (with credentials, if any) inside /etc/config/network and your voip inside /etc/config/mmpbx* (with some adjustments inside /etc/config/mwan and /etc/config/dhcp). You will need to change as much options as much your ISP settings differs from Telia ones.
> 
> Let me close this issue as rooting support for VANT-R is now complete. Please, open a new issue for "Help configuring VANT-R for WE" if needed, sharing every useful knowledge you already have about your ISP settings.
> 
> Thank you for your help.


Glad I could take part in the process even If I didn't do much brain work.
I'll keep the router next to me for few days.
I'll check from time to time about custom GUI.
I appreciate your time and effort.
